### PR TITLE
[RFR] Dockerfile

### PR DESCRIPTION
**Pull request by sshveta**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+ENV WORK $PROJECTS/$RHAMTENV
+
+RUN mkdir $PROJECTS && cd $PROJECTS
+
+WORKDIR $PROJECTS
+
+RUN virtualenv $RHAMTENV
+
+WORKDIR $RHAMTENV
```
Comment (digitronik) [inline Dockerfile:61]: why we need this?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+MAINTAINER Shveta Sachdeva <sshveta@redhat.com>
+LABEL description="Red Hat Application Migration Toolkit Web-Console"
+
+RUN dnf -y update && dnf clean all
+RUN dnf -y install java java-devel unzip && dnf clean all
+
+# set java env
+ENV JAVA_HOME /usr/lib/jvm/java-openjdk
+
+wget WEB_CONSOLE_FILE_PATH /tmp
```
Comment (digitronik) [inline Dockerfile:12]: I don't this it will work... ^^

File: main.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
+BASE_URL = "https://repo1.maven.org/maven2/org/jboss/windup"
```
Comment (digitronik) [inline main.py:1]: Q: why `main.py`?

**Comment by sshveta** [comment]
> I think we should test it first locally. I have some doubts like
> 
>     * no base file
> 
>     * Its looks like we are trying to do containerised testing not containerised RHAMT application

The new dockerfile is working fine on jenkins and is successfully creating the build .

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+ENV BASE_URL="https://repo1.maven.org/maven2/org/jboss/windup"
+ENV WEB_CONSOLE="rhamt-web-distribution"
+ENV VERSION="4.3.1.Beta1"
+ENV WEB_DISTRIBUTION="${WEB_CONSOLE}-${VERSION}"
+ENV WEB_CONSOLE_FILE="${WEB_DISTRIBUTION}-with-authentication.zip"
+ENV WEB_CONSOLE_FILE_PATH="${BASE_URL}/web/${WEB_CONSOLE}/${VERSION}/${WEB_CONSOLE_FILE}"
+
+
+# https://repo1.maven.org/maven2/org/jboss/windup/web/rhamt-web-distribution/4.3.1.Beta1/rhamt-web-distribution-4.3.1.Beta1-with-authentication.zip
+# https://repo1.maven.org/maven2/org/jboss/windup/rhamt-cli/4.3.1.Beta1/rhamt-cli-4.3.1.Beta1-offline.zip
```
Comment (digitronik) [inline Dockerfile:21]: Please remove these comment lines.

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+RUN echo $WEB_CONSOLE_FILE_PATH
```
Comment (digitronik) [inline Dockerfile:23]: Please remove this line; I think you added just of check.

File: Dockerfile [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+RUN wget -o - $WEB_CONSOLE_FILE_PATH -P /tmp| wc -l > /number
+RUN unzip -o /tmp/$WEB_CONSOLE_FILE -d /tmp
```
Comment (digitronik) [inline Dockerfile:26]: Can we combine these two; we can try to reduce layes.

```
RUN wget -o - $WEB_CONSOLE_FILE_PATH -P /tmp| wc -l > /number && unzip -o /tmp/$WEB_CONSOLE_FILE -d /tmp
```

File: Dockerfile [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+RUN useradd jboss \
+        && usermod -G jboss jboss \
+        && mkdir /opt/rhamt \
+        && mkdir /opt/submitted-ears \
+        && mkdir /opt/rhamt-data
+
+WORKDIR /opt/rhamt
+
+RUN cp -r /tmp/${WEB_DISTRIBUTION}/* ./ && chown jboss:jboss /opt -R
```
Comment (digitronik) [inline Dockerfile:36]: In state of copy we can move it.

File: Dockerfile (hunk 4 above)
Comment (sshveta) [reply Dockerfile:21]: Hi , I want to keep these as example of repos used in downloading snapshots

**Pull request merged**